src/components/filterslider.js

```diff
@@ -0,0 +1,109 @@
+import React, { Component } from 'react'
+import {clothingCategory,stylesByCategory,clothingColor,clothingSeason} from '../containers/api'
+import  '../styles/filterslider.css'
+
+
+export default class Filter extends Component {
+    constructor(props) {
+        super(props)
+        this.state = {
+            categories: clothingCategory,
+            chosencategory: '',
+            styles: [],
+            chosenstyle: '',
+            colors: clothingColor,
+            chosencolor: '',
+            seasons: clothingSeason,
+            chosenseason: ''
+        }
+    }
+
+
+    //---------HANDLES SELECTIONS 
+    handleCategory = (e) => {
+      this.setState({
+          chosencategory: e.target.value,
+          styles: stylesByCategory[e.target.value]
+        })
+      
+}
+
+    handleStyle = (e) => {
+        console.log('STYLE selected', e.target.value);
+        this.setState({ chosenstyle: e.target.value })
+
+        
+    }
+
+    handleColor = (e) => {
+        console.log('COLOR selected', e.target.value);
+        this.setState({ chosencolor: e.target.value })
+    }
+
+    handleSeason = (e) => {
+        console.log('SEASON selected', e.target.value);
+        this.setState({ chosenseason: e.target.value })
+    }
```

Comment:
You can DRY up your code here by combining the `handle*` functions into a single `handleSelection` function. It might look something like this:
```
handleSelection = (e, field) => {
  this.SetState({ [field]: e.target.value });
}
```
To make this work, you will also need to change the callbacks you pass to your `<select >` elements' `onChange` attributes.
```
<select onChange={ (e) => { this.handleSelection(e, 'chosencategory'); } } >
<select onChange={ (e) => { this.handleSelection(e, 'chosenstyle'); } } >
etc.

```